RELEASING.md

```diff
@@ -1,16 +1,21 @@
 # Releasing
 
-> **Warning**
-> It's important to always do an engine release prior to releasing any SDK. This will ensure that all the APIs in the SDK are also available in the engine it depends on.
+This document describes how to release:
+- 🚙 Dagger Engine & CLI `v0.3.x`
+- 🐹 Go SDK
+- 🐍 Python SDK
+- ⬡ Node.js SDK
 
-> **Warning**
-> Currently, if any SDK bumps an engine dependency, we should bump the engine dependency of all other SDKs concurrently. This helps ensure that users running multiple SDKs from a single machine all use the same engine. This is important as currently our automatic provisioning code enforces the existence of a single engine running at a time.
```

Comment:
This deleted warning about upgrading SDKs in lockstep with one another is still true, so I think it should still exist in this doc in one form or another. It's not a technical hard requirement but it's greatly preferable for the reason given in the warning.

Reply:
I was thinking the same thing, but then I realised that:

<img width="939" alt="image" src="https://user-images.githubusercontent.com/3342/205131736-0454db49-6d1d-481c-94de-cd068374007b.png">

Would result in this:

<img width="903" alt="image" src="https://user-images.githubusercontent.com/3342/205131815-2f4ebca6-e285-460b-b87a-00a6690ec517.png">

Which means that we should **not** be bumping the Engine version in SDKs manually.

If anything, we should add a pre-publish check that would fail the publish if the Engine version differs between SDKs.

Reply:
That's true, though it's still possible to merge those PRs with the dependency bump but then not actually do a release of the SDK. The important part is that not only are the engine dependencies bumped in each SDK in lockstep, each SDK should also ideally be published in lockstep. So I think it's still worth noting that somewhere.

> If anything, we should add a pre-publish check that would fail the publish if the Engine version differs between SDKs. WDYT?

Yeah that seems like a good idea.

On a related note, I just sent out a PR a few mins ago to check that the latest released engine has an API that matches the one used by the SDKs: https://github.com/dagger/dagger/pull/4065

I think that will work well in concert with this.

Reply:
Resolved in follow-up change. The warning is now back.